From: =?UTF-8?q?Miloslav=20Trma=C4=8D?= <mitr@redhat.com>
Date: Sat, 13 Apr 2013 03:19:17 +0200
Subject: [PATCH 1/3] Fix various memory leaks.

The XML_ParserCreate_MM one was found by Florian Weimer in
https://bugzilla.redhat.com/show_bug.cgi?id=888728 .

This should cover everything found by valgrind on the JS authority
test, augmented with a call to
polkit_backend_authority_enumerate_actions() to verify the
XML_ParserCreate_MM case.

https://bugs.freedesktop.org/show_bug.cgi?id=63492
---
 src/netgroup.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/netgroup.c b/src/netgroup.c
index f2ee857..bc99555 100644
--- a/src/netgroup.c
+++ b/src/netgroup.c
@@ -122,7 +122,10 @@ struct netgroup *netgroup_parse_all() {
     char * line = NULL;
     ssize_t line_size = getline(&line, &line_alloc, stream);
     if (line_size == -1)
-      break;
+      {
+	free(line);
+	break;
+      }
 
     struct netgroup *nextgroup = netgroup_parse_line(line);
     free(line);
-- 
2.43.0

